NumLib/NamedFunctionCaller.h

```diff
@@ -0,0 +1,133 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_NAMEDFUNCTIONCALLER_H
+#define NUMLIB_NAMEDFUNCTIONCALLER_H
+
+#include <map>
+#include <vector>
+#include "NamedFunction.h"
+
+namespace NumLib
+{
+class SpecialFunctionCaller;
+
+//! Builds expression trees of named functions dynamically at runtime.
+class NamedFunctionCaller final
+{
+public:
+    //! Constructs an instance whose unbound arguments have the given names.
+    explicit NamedFunctionCaller(
+        std::initializer_list<std::string> unbound_arguments);
+
+    //! Adds the given named function
+    void addNamedFunction(NamedFunction&& fct);
```

Comment:
I think that wouldn't be handy in the end. Currently I build a `NamedFunctionCaller` in the constructor of `Process` and add functions to it later on during the initialization.
Passing all functions to the ctor would mean that everything has to be ready from the beginning.